src/libraries/System.Net.HttpListener/tests/HttpListenerTests.cs

```diff
@@ -155,5 +156,30 @@ public void EndGetContext_AlreadyCalled_ThrowsInvalidOperationException()
                 Assert.Throws<InvalidOperationException>(() => listener.EndGetContext(beginGetContextResult));
             }
         }
+
+        [Fact]
+        [OuterLoop]
+        public void GetContext_StopIsCalled_GetContextUnblocked()
+        {
+            using var listenerFactory = new HttpListenerFactory();
+            var listener = listenerFactory.GetListener();
+            listener.Start();
+
+            var listenerTask = Task.Run(() =>
+            {
+                try
+                {
+                    var context = listener.GetContext();
+                }
+                catch (Exception)
+                {
+                }
+            });
+
+            Task.Delay(1000).Wait(1100); // let listenerTask to call GetContext
```

Comment:
Can this method be made async so you can `await` instead of `.Wait()`? If not, `Thread.Sleep` would be better.